spring-data-jdbc/src/test/java/org/springframework/data/jdbc/core/PersistentPropertyPathExtensionUnitTests.java

```diff
@@ -130,12 +132,69 @@ public void getColumnName() {
 
 		SoftAssertions.assertSoftly(softly -> {
 
-			softly.assertThat(extPath("second.third.value").getColumnName()).isEqualTo("thrdvalue");
-			softly.assertThat(extPath("second.third2.value").getColumnName()).isEqualTo("value");
-			softly.assertThat(extPath("secondList.third.value").getColumnName()).isEqualTo("thrdvalue");
-			softly.assertThat(extPath("secondList.third2.value").getColumnName()).isEqualTo("value");
-			softly.assertThat(extPath("second2.third.value").getColumnName()).isEqualTo("secthrdvalue");
-			softly.assertThat(extPath("second2.third2.value").getColumnName()).isEqualTo("value");
+			softly.assertThat(extPath("second.third2.value").getColumnName()).isEqualTo("thrdvalue");
+			softly.assertThat(extPath("second.third.value").getColumnName()).isEqualTo("value");
+			softly.assertThat(extPath("secondList.third2.value").getColumnName()).isEqualTo("thrdvalue");
+			softly.assertThat(extPath("secondList.third.value").getColumnName()).isEqualTo("value");
+			softly.assertThat(extPath("second2.third2.value").getColumnName()).isEqualTo("secthrdvalue");
+			softly.assertThat(extPath("second2.third.value").getColumnName()).isEqualTo("value");
+		});
+	}
+
+	@Test // DATAJDBC-359
+	public void idDefiningPath() {
+
+		SoftAssertions.assertSoftly(softly -> {
```

Comment:
Let's statically import `assertSoftly`. These deeply indented blocks make it hard to read.